docs/utilities/data_masking.md

```diff
@@ -0,0 +1,587 @@
+
+#### JSON
+
+We also support data in JSON string format as input. We automatically deserialize it, then handle each field operation as expected.
+
+Note that the return will be a deserialized JSON and your desired fields updated.
+
+=== "Data"
+
+    Expression: `data_masker.erase(data, fields=["card_number", "address.postcode"])`
+
+    ```json
+    --8<-- "examples/data_masking/src/choosing_payload_simple_json.json"
+    ```
+
+=== "Result"
+
+    ```json
+    --8<-- "examples/data_masking/src/choosing_payload_simple_json_output.json"
+    ```
+
+## Advanced
+
+### Data serialization
+
+???+ note "Current limitations"
+    1. Python classes, `Dataclasses`, and `Pydantic models` are not supported yet.
+
+Before we traverse the data structure, we perform two important operations on input data:
+
+1. If `JSON string`, **deserialize** using default or provided deserializer.
+2. If `dictionary`, **normalize** into `JSON` to prevent traversing unsupported data types.
+
+When decrypting, we revert the operation to restore the original data structure.
+
+For compatibility or performance, you can optionally pass your own JSON serializer and deserializer to replace `json.dumps` and `json.loads` respectively:
+
+=== "advanced_custom_serializer.py"
+
+    ```python hl_lines="17-18"
+    --8<-- "examples/data_masking/src/advanced_custom_serializer.py"
+    ```
+
+### Providers
+
+#### AWS Encryption SDK
+
+You can modify the following values when initializing the `AWSEncryptionSDKProvider` to best accommodate your security and performance thresholds.
+
+| Parameter                  | Default               | Description                                                                                   |
+| -------------------------- | --------------------- | --------------------------------------------------------------------------------------------- |
+| **local_cache_capacity**   | `100`                 | The maximum number of entries that can be retained in the local cryptographic materials cache |
+| **max_cache_age_seconds**  | `300`                 | The maximum time (in seconds) that a cache entry may be kept in the cache                     |
+| **max_messages_encrypted** | `4294967296`          | The maximum number of messages that may be encrypted under a cache entry                      |
+| **max_bytes_encrypted**    | `9223372036854775807` | The maximum number of bytes that may be encrypted under a cache entry                         |
+
+**Changing the default algorithm**
+
+The AWS Encryption SDK defaults to using the `AES_256_GCM_HKDF_SHA512_COMMIT_KEY_ECDSA_P384` algorithm for encrypting your Data Key. If you want, you have the flexibility to customize and choose a different encryption algorithm.
+
+=== "changing_default_algorithm.py"
+
+    ```python hl_lines="5 26"
+    --8<-- "examples/data_masking/src/changing_default_algorithm.py"
+    ```
```

Comment:
For completion, we need to explain how to override `EncryptionSDKClient` options (algorithm is one of them).

suggestion: rename to `Passing additional SDK arguments`, and keep the `algorithm` as one of the arguments they can use via `provider_options`.

<img width="1021" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/3340292/53c13d24-0ab3-4bc1-83cd-1d78fd697182">